source/include/FreeRTOS_Routing.h

```diff
@@ -248,13 +248,21 @@
                                                         uint32_t ulWhere );
 
 /*
- * Find the best fitting end-point to reach a given IP-address on a given interface
+ * Find the best fitting IPv4 end-point to reach a given IP-address on a given interface
  * 'ulWhere' is temporary and or debugging only.
  */
     NetworkEndPoint_t * FreeRTOS_InterfaceEndPointOnNetMask( const NetworkInterface_t * pxInterface,
                                                              uint32_t ulIPAddress,
                                                              uint32_t ulWhere );
 
+/*
+ * Find the best fitting IPv6 end-point to reach a given IP-address on a given interface
+ * 'ulWhere' is temporary and or debugging only.
+ */
+    NetworkEndPoint_t * FreeRTOS_InterfaceEPOnNetMask_IPv6( const NetworkInterface_t * pxInterface,
```

Comment:
Finds an endpoint on the given interface which is in the same subnet as the
given IP address. If NULL is passed for pxInterface, it looks through all the
interfaces to find an endpoint in the same subnet as the given IP address.

FreeRTOS_FindInterfaceEndPointInSameSubnet_IPv6( const NetworkInterface_t * pxInterface,
                                                 const IPv6_Address_t * pxIPAddress )
{

}